app/controllers/UpfrontPaymentController.scala

```diff
@@ -87,23 +106,24 @@ class UpfrontPaymentController @Inject() (
       hasUpfrontPayment = Some(true),
       upfrontAmount     = Some(AmountInPence(10000L))
     )
-    Future.successful(Ok(upfrontSummaryPage(mockUserAnswers, AmountInPence(200000L))))
+    Future.successful(Ok(views.upfrontSummaryPage(mockUserAnswers, AmountInPence(200000L))))
   }
 }
 
 object UpfrontPaymentController {
-  val key: String = "UpfrontPaymentAmount"
 
-  def upfrontPaymentForm(): Form[String] = Form(
+  def canYouMakeAnUpfrontPaymentQuestionForm: Form[String] = Form(
```

Comment:
we wouldn't need `formToCanPayUpfront` if we did some sort of transformation here and had `Form[Boolean]` or `Form[CanPayUpfront]`instead.

Reply:
will make this change now

Reply:
This one is tricky and important.

We don't want to rely on play's default messages which requires us to provide some translations in messages files. We'd be more happy if we could pass required messages directly into form formatters, like here:
https://github.com/hmrc/essttp-frontend/blob/636ba084caebf665a4ffad0a9383666d8ae58d08/app/testOnly/formsmodel/StartJourneyForm.scala#L42

So we need to figure out how to pass required messages to the form validator if the field is missing. 
I'll create an example in a moment.

Reply:
https://github.com/hmrc/essttp-frontend/pull/48